lmdb is not optional if using something that relies on `@angular/build/private`

### Command

build

### Is this a regression?

- [x] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

20.3.13

### Description

### Some context before:

Through this refactoring:

https://github.com/angular/angular-cli/commit/5e29b4432371d425819d32fd1e3d9befad2bb455

The `lmdb-cache-store.ts` file is now being re-exported. 

When transpiled (see https://www.npmjs.com/package/@angular/build/v/21.0.3?activeTab=code):

`var lmdb_cache_store_1 = require("./tools/esbuild/lmdb-cache-store");`

Now, `lmbd-cache-store.ts` imports `lmdb` (which later on will be `require('lmdb')`):

https://github.com/angular/angular-cli/blob/efd1180d7bf80471f729e6122b180a895faa13b0/packages/angular/build/src/tools/esbuild/lmdb-cache-store.ts#L9

### Where the problem starts:

Other files within this repository later on import stuff from `@angular/build/private`. Just as an example:

https://github.com/angular/angular-cli/blob/v21.0.3/packages/angular_devkit/build_angular/src/builders/server/index.ts#L9

Which ends up being transpiled as (https://www.npmjs.com/package/@angular-devkit/build-angular/v/21.0.3?activeTab=code):

`const private_1 = require("@angular/build/private");`

So, assuming a simple script such as:

```js
const devkit = require("@angular-devkit/build-angular");

const executeServerBuilder = devkit.executeServerBuilder;
```

Will fail when run if `lmdb` is not installed (because of the chain of _imports / requires_). 

### Minimal Reproduction

Create a script such as the one mentioned above: 

```js
// index.js
const devkit = require("@angular-devkit/build-angular");

// Or anything that ends up loading `@angular/build/private`.
const executeServerBuilder = devkit.executeServerBuilder;
```

1. Start a new npm package or similar: `npm init -y`
2. Install `@angular-devkit/build-angular`, **without installing the optional dependencies**: `npm i @angular-devkit/build-angular@21.0.3 --save-dev --omit optional`.
3. Run the previous script, e.g. `node .` or `node index.js`.

### Exception or Error

```text
node:internal/modules/cjs/loader:1386
  throw err;
  ^

Error: Cannot find module 'lmdb'
```

### Your Environment

```text
_                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI       : 21.0.3
Node.js           : 22.19.0
Package Manager   : npm 10.9.3
Operating System  : darwin arm64

┌───────────────────────────────┬───────────────────┬───────────────────┐
│ Package                       │ Installed Version │ Requested Version │
├───────────────────────────────┼───────────────────┼───────────────────┤
│ @angular-devkit/build-angular │ 21.0.3            │ ^21.0.3           │
│ @angular/cli                  │ 21.0.3            │ ^21.0.3           │
└───────────────────────────────┴───────────────────┴───────────────────┘
```

### Anything else relevant?

_No response_

--- a/packages/angular/build/src/private.ts
+++ b/packages/angular/build/src/private.ts
@@ -36,7 +36,6 @@ export { SassWorkerImplementation } from './tools/sass/sass-service';
 
 export { SourceFileCache } from './tools/esbuild/angular/source-file-cache';
 export { Cache } from './tools/esbuild/cache';
-export { LmdbCacheStore } from './tools/esbuild/lmdb-cache-store';
 export { createJitResourceTransformer } from './tools/angular/transformers/jit-resource-transformer';
 export { JavaScriptTransformer } from './tools/esbuild/javascript-transformer';
 

